### PR TITLE
remove dependency from go-circom-prover-verifier

### PR DESCRIPTION
**Pull request by ilya-korotya**

### CODE REVIEW

File: internal/models/models.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+package models
+
+import (
+	"math/big"
+
+	bn256 "github.com/ethereum/go-ethereum/crypto/bn256/cloudflare"
+)
+
+// R is the mod of the finite field
+var R, _ = new(big.Int).SetString("21888242871839275222246405745257275088548364400416034343698204186575808495617", 10)
```
Comment (vmidyllic) [inline internal/models/models.go:12]: move this number to constant or use constant from go-iden3-crypto lob

File: internal/models/models.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+// ProofData describes three components of zkp proof in bn256 format.
+type ProofData struct {
```
Comment (vmidyllic) [inline internal/models/models.go:15]: it's better to rename it to ProofPairingData and why do we need internal/models ?

Comment (vmidyllic) [inline internal/models/models.go:15]: it's better to rename it to ProofPairingData and why do we need internal/models ?

File: types/proof.go [hunk 3]
```diff
@@ -48,6 +55,30 @@ type ProofData struct {
 	Protocol string     `json:"protocol"`
 }
 
+func (pr *ProofData) ToInternalProofData() (models.ProofData, error) {
```
Comment (vmidyllic) [inline types/proof.go:58]: let's refactor models.ProofData. Quite confusing method

Comment (vmidyllic) [inline types/proof.go:58]: let's refactor models.ProofData. Quite confusing method

File: types/proof.go [hunk 4]
```diff
@@ -90,3 +121,60 @@ type SignatureProofRequest struct {
...
+// VkString is the Verification Key data structure in string format (from json).
+type VkString struct {
+	Alpha []string   `json:"vk_alpha_1"`
+	Beta  [][]string `json:"vk_beta_2"`
+	Gamma [][]string `json:"vk_gamma_2"`
+	Delta [][]string `json:"vk_delta_2"`
+	IC    [][]string `json:"IC"`
+}
+
+func (vk *VkString) ToInternalVk() (*models.Vk, error) {
```
Comment (vmidyllic) [inline types/proof.go:134]: to VK is better.

Comment (vmidyllic) [inline types/proof.go:134]: to VK is better.

File: verification/proof.go [hunk 5]
```diff
@@ -3,44 +3,36 @@ package verification
...
-	p, err := parsers.ParseProof(proofBytes)
-	if err != nil {
-		return err
-	}
-
-	// 2. parse inputs to [] string if needed
 
-	vkKey, err := parsers.ParseVk(verificationKey)
+	// 2. cast external verification key data to internal model.
+	var vk types.VkString
```
Comment (vmidyllic) [inline verification/proof.go:24]: var vkStr types.VkString to not confuse.

Comment (vmidyllic) [inline verification/proof.go:24]: var vkStr types.VkString to not confuse.

File: internal/models/models.go (hunk 2 above)
Comment (ilya-korotya) [reply internal/models/models.go:15]: I want to hide internal structures from the end user. Because he doesn't need them.

**Review by demonsh — APPROVED** [review]

**Pull request merged**